### PR TITLE
feat(ctb): Pass `msg.sender` to dispute games from the factory

### PR DESCRIPTION
**Pull request by clabby**
## Overview

Passes `msg.sender` from the factory to the dispute game as an immutable argument. This change allows for smart wallets to participate in the protocol as first-class citizens.

**Metadata**
closes https://github.com/ethereum-optimism/client-pod/issues/732
closes https://github.com/ethereum-optimism/client-pod/issues/731

### CODE REVIEW

**Comment by clabby** [comment]
* **#10154** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/10154?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#10153** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/10153?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#10152** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/10152?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#10151** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/10151?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#10150** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/10150?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#10149** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/10149?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#10148** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/10148?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @clabby and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/contracts-bedrock/src/dispute/interfaces/IFaultDisputeGame.sol [hunk 1]
```diff
@@ -62,9 +62,6 @@ interface IFaultDisputeGame is IDisputeGame {
     /// @param _claimIndex The index of the subgame root claim to resolve.
     function resolveClaim(uint256 _claimIndex) external payable;
 
-    /// @notice A block hash on the L1 that contains the disputed output root.
```
Comment (refcell) [inline packages/contracts-bedrock/src/dispute/interfaces/IFaultDisputeGame.sol:65]: nit
```suggestion
    /// @notice A block hash on L1 that contains the disputed output root.
```

**Review by refcell — APPROVED** [review]

**Review by smartcontracts — APPROVED** [review]

File: packages/contracts-bedrock/src/dispute/interfaces/IFaultDisputeGame.sol (hunk 1 above)
Comment (clabby) [reply packages/contracts-bedrock/src/dispute/interfaces/IFaultDisputeGame.sol:65]: Was changed in the `IDisputeGame` interface 👍

**Comment by laizy** [comment]
@smartcontracts, Sorry to bother you, but I reported this same issue in https://github.com/sherlock-audit/2024-02-optimism-2024-judging/issues/140 as https://github.com/sherlock-audit/2024-02-optimism-2024-judging/issues/194 and didn't receive the reward. hope you could take a look, and contact to the Sherlock team. I'm not specialized in auditing, and this is my first time participating. I joined this competition because I'm passionate about optimism. I've been studying optimism contract code for years and have learned a lot from you!

**Pull request merged**